server/chat-commands.js

```diff
@@ -2803,9 +2805,22 @@ const commands = {
 		let name = this.targetUsername;
 		if (!targetUser && !room.log.hasUsername(target)) return this.errorReply(`User ${target} not found or has no roomlogs.`);
 		let userid = toID(this.inputUsername);
-		if (!this.can('mute', null, room)) return;
 
-		if (targetUser && (cmd === 'hidealtstext' || cmd === 'hidetextalts' || cmd === 'hidealttext')) {
+		if (!user.can('mute', targetUser, room)) {
+			if (!this.can('mute', null, room)) return;
+			if (targetUser !== user) {
+				if (!cmd.includes('force')) {
+					return this.errorReply(`${target} is a higher rank than you, are you sure you want to hide their messages? Use /forcehidetext if you're sure.`);
+				}
+				// Notify staff when someone's abusing hidetext in a public room
```

Comment:
I think this should be rephrased (to at least include the word 'potentially') - simply using `forcehidetext` is not necessarily 'abuse', or why would we even be allowing it?